metadatastore/core.py

```diff
@@ -1043,7 +1044,7 @@ def find_events(start_col, start_cache,
         # wrap it our fancy dict
         ev = doc.Document('Event', ev)
         yield ev
-
+    events.close()
```

Comment:
If a cursor is closed and `close()` method is called on this closed cursor, no exception is raised. 

```
n [5]: conn.test.mock.find()
Out[5]: <pymongo.cursor.Cursor at 0x104907748>

In [6]: a = conn.test.mock.find()

In [7]: a.close()

In [8]: a.close()
```
 
https://github.com/mongodb/mongo-python-driver/blob/master/pymongo/cursor.py#L276